Detectors/TPC/simulation/src/Detector.cxx

```diff
@@ -140,6 +142,28 @@ Bool_t Detector::ProcessHits(FairVolume* vol)
   // TODO: Temporary hack to process only one sector
   // if (sectorID != 0) return kFALSE;
 
+  // ---| remove clusters between the IFC and the FC strips |---
+  // those should not enter the active readout area
+  // do coarse selection before, to limit number of transformations
+  if (detParam.ExcludeFCGap) {
+    const auto rCluster = std::sqrt(position.X() * position.X() + position.Y() * position.Y());
+    const float rodRin = 81.5 + 2.2;    // radial position of the inner field cage rods + radial size of the field cage rods
+    const float rodRout = 254.25 + 2.2; // radial position of the outer field cage rods + radial size of the field cage rods
+    const float fcLxIn = 82.428409;     // position of the inner FC strips in local x = cos(10 deg) * rodRin;
+    const float fcLxOut = 252.55395;    // position of the outer FC strips in local x = cos(10 deg) * rodRin;
+
+    if (rCluster < rodRin || rCluster > fcLxOut) {
+      const int sectorIDnonShift = static_cast<int>(Sector::ToSector(position.X(), position.Y(), position.Z()));
+      const double alpha = TMath::DegToRad() * (10. + sectorIDnonShift * 20.);
+      const double cs = std::cos(-alpha), sn = std::sin(-alpha);
+      const auto localX = position.X() * cs - position.Y() * sn;
+      // fine cut
+      if (localX < fcLxIn || localX > fcLxOut) {
+        return kFALSE;
+      }
+    }
+  }
```

Comment:
ok. So this is removing things even on the hit level, right?

Reply:
Yes, in real life they should not enter the active volume. This was no problem before, since with didn't simulate distortions. But with distortions they would end up in the active area, which physically does not make sense.